bundles/org.openhab.binding.liquidcheck/src/main/java/org/openhab/binding/liquidcheck/internal/LiquidCheckHandler.java

```diff
@@ -0,0 +1,181 @@
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link LiquidCheckHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Goerentz - Initial contribution
+ */
+@NonNullByDefault
+public class LiquidCheckHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(LiquidCheckHandler.class);
+    private final HttpClient httpClient;
+
+    private Map<String, String> oldProps = new HashMap<>();
+
+    private LiquidCheckConfiguration config = new LiquidCheckConfiguration();
+    private LiquidCheckHttpClient client = new LiquidCheckHttpClient(config, new HttpClient());
+
+    private @Nullable ScheduledFuture<?> polling;
+
+    public LiquidCheckHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(MEASURE_CHANNEL)) {
+            if (command instanceof OnOffType) {
+                try {
+                    if (client != null && client.isConnected()) {
+                        String response = client.measureCommand();
+                        CommData commandResponse = new Gson().fromJson(response, CommData.class);
+                        if (commandResponse != null) {
+                            if (!"success".equals(commandResponse.context.status)) {
+                                logger.warn("Starting the measurement was not successful!");
+                            }
+                        } else {
+                            logger.warn("The object commandResponse is null!");
+                        }
+                    }
+                } catch (TimeoutException | ExecutionException | JsonSyntaxException e) {
+                    logger.warn("This went wrong in handleCommand: {}", e.getMessage());
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+                updateState(channelUID, OnOffType.OFF);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(LiquidCheckConfiguration.class);
+        oldProps = thing.getProperties();
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
+
+            this.client = new LiquidCheckHttpClient(config, httpClient);
+            boolean thingReachable = this.client.isConnected();
+            if (thingReachable) {
+                updateStatus(ThingStatus.ONLINE);
+                PollingForData pollingRunnable = new PollingForData(this.client);
+                polling = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, config.refreshInterval,
+                        TimeUnit.SECONDS);
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
```

Comment:
```suggestion
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Additionally, this seems to be a single attempt. So the binding will not be able to recover if there are network issues during initialization?

Reply:
Yes, it is a single attempt, because I think if there is an issue with the connection this won't change within the next attempts. And the user should have a look at it first before retrying.

If you want me to change it, it will not be an issue 😄

Reply:
Okay, I understand.
So should there be a limit to the number of retries or to the time?
Or will there be endless retries until the initialization has been completed?

Reply:
A variety of different implementations is possible, some more sophisticated than others. For this purpose probably it would suffice to retry infinitely with the **refreshInterval** frequency.

Reply:
Okay, got it.